internal/pkg/agent/application/secret/secret_test.go

```diff
@@ -0,0 +1,71 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+//go:build linux || windows
+// +build linux windows
+
+package secret
+
+import (
+	"os"
+	"path/filepath"
+	"testing"
+	"time"
+
+	"github.com/elastic/elastic-agent/internal/pkg/agent/vault"
+	"github.com/google/go-cmp/cmp"
+)
+
+func getTestVaultPath() string {
+	exe, _ := os.Executable()
+	dir := filepath.Dir(exe)
+	return filepath.Join(dir, "vault", "co.elastic.agent")
+}
```

Comment:
all tests share the same dir here? is that what we want?
Maybe you should make this a helper function and use `t.TempDir` to generate a directory for each test case

Reply:
sure, updated